src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -149,14 +162,28 @@ public Secret getPassword() {
             apiUri = "https://api.github.com";
         }
 
-        long now = System.currentTimeMillis();
+        long now = Instant.now().getEpochSecond();
         String appInstallationToken;
-        if (cachedToken != null && now - tokenCacheTime < JwtHelper.VALIDITY_MS /* extra buffer */ / 2) {
-            appInstallationToken = cachedToken;
-        } else {
-            appInstallationToken = generateAppInstallationToken(appID, privateKey.getPlainText(), apiUri, owner);
-            cachedToken = appInstallationToken;
-            tokenCacheTime = now;
+        synchronized (this) {
```

Comment:
If multiple requests for a password are made concurrently, we want to just get one refreshed token.